packages/sharing-smart-contract/test/collection.test.js

```diff
@@ -131,9 +137,20 @@ describe('Collection', () => {
       const protectedDataAddress = await createDatasetForContract(addr1.address, rpcURL);
       const tx = protectedDataSharingContract
         .connect(addr1)
-        .addProtectedDataToCollection(collectionTokenId, protectedDataAddress);
+        .addProtectedDataToCollection(collectionTokenId, protectedDataAddress, TEST_APP_ADDRESS);
+
+      await expect(tx).to.be.revertedWith('ProtectedDataSharing Contract not approved');
+    });
+    it('Should revert if app address is invalid', async () => {
+      const { protectedDataSharingContract, collectionTokenId, addr1 } =
+        await loadFixture(createCollection);
+
+      const protectedDataAddress = await createDatasetForContract(addr1.address, rpcURL);
+      const tx = protectedDataSharingContract
+        .connect(addr1)
+        .addProtectedDataToCollection(collectionTokenId, protectedDataAddress, ethers.ZeroAddress);
 
-      await expect(tx).to.be.revertedWith('Collection Contract not approved');
+      await expect(tx).to.be.revertedWith('App address invalid');
```

Comment:
see the comment on the require, we should be more restrictive

Reply:
Commit: 8cd1186ee574f5d2f8ca006f7d70d68f4160f89d
Fix